### PR TITLE
migrate 'lib/RuleSet.js' to es6

### PR DESCRIPTION
**Pull request by Tushkiz**
**What kind of change does this PR introduce?**
Refactor

**Did you add tests for your changes?**
N/A

**If relevant, link to documentation update:**
N/A

**Summary**
Refactors 'lib/RuleSet.js' to es6

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: lib/RuleSet.js [hunk 1]
```diff
@@ -69,277 +69,340 @@ normalized:
...
+			rule.use.forEach((use) => {
+				result.push({
+					type: "use",
+					value: typeof use === "function" ? RuleSet.normalizeUseItemFunction(use, data) : use,
+					enforce: rule.enforce
+				});
+			});
+		}
+
+		let i;
```
Comment (sokra) [inline lib/RuleSet.js]: move this into the for loops: `for(let i = 0`...

File: lib/RuleSet.js [hunk 2]
```diff
@@ -69,277 +69,340 @@ normalized:
...
+		return newItem;
 	}
-	if(typeof condition === "function") {
-		return condition;
+
+	static normalizeCondition(condition) {
+		if(!condition)
+			throw new Error("Expected condition but got falsy value");
+		if(typeof condition === "string") {
+			return (str) => str.indexOf(condition) === 0;
```
Comment (sokra) [inline lib/RuleSet.js]: remove brackets here elsewise the linter doesn't like it: `return str => str ...`

Comment (Tushkiz) [inline lib/RuleSet.js]: done!

**Comment by Tushkiz** [comment]
Hey @sokra, do you want me to rebase and update this PR for easy merger.

**Comment by sokra** [comment]
Thanks

**Pull request merged**